drivers/lsm6ds3/lsm6ds3.c

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (C) 2015 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_lsm6ds3
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the LSM6DS3 3D accelerometer
+ *              and 3D gyroscope
+ *
+ * @author      René Herthel <rene-herthel@outlook.de>
+ *
+ * @}
+ */
+
+#include "lsm6ds3.h"
+#include "include/lsm6ds3-internal.h"
+
+#define ENABLE_DEBUG       (0)
+#include "debug.h"
+
+#define TEMP_SENSI         (16)
+#define TEMP_OFFSET        (25)
+#define MASK_CHAR_MSB      (0x8000)
+#define MASK_CHAR_NOT_MSB  (0x7FFF)
+
+/**
+ * @brief Calculate the two's complement of a value
+ */
+static inline char _twos_complement(char value, int mask)
+{
+    return ~(~(value & mask) + 1);
```

Comment:
a char in C is 1 byte, your masks have 2 bytes defined. are you sure about this calculation?